primer/src/Primer/App/Base.hs

```diff
@@ -70,7 +72,7 @@ data NodeType = BodyNode | SigNode
 -- | Describes which element of a (type or term) definition the student has selected.
 -- We have the following invariant: when this contains a `NodeSelection` with @nodeType = SigNode@,
 -- or any `TypeDefConsFieldSelection`, then they will always have @meta = Right _@.
-type Selection = Selection' (Either ExprMeta TypeMeta)
+type Selection = Selection' (Either ExprMeta (Either TypeMeta KindMeta))
```

Comment:
This nested `Either` is maybe a bit ugly, but it's still in some ways simpler than defining a new type, whilst still being fairly self-explanatory.